npm/hermes-engine-darwin/unpack-tarball.js

```diff
@@ -0,0 +1,19 @@
+const path = require("path");
+const fs = require("fs");
+const child_process = require("child_process");
+
+if (
+  process.platform === "darwin" &&
+  !fs.existsSync(path.join(__dirname, "destroot"))
+) {
+  const tarball = fs.readdirSync(__dirname).find(function (entry) {
+    return entry.endsWith(".tar.gz");
```

Comment:
Can we also check startsWith("hermes-runtime-darwin-"), just in case something else ends up in here?

Reply:
I figured keeping this flexible would make it future-proof against renaming the tarball, as this happens in a different build job, but you’re right that it might be bad if some other artefact that this script doesn’t expect got included too.